backend/t4cclient/sessions/routes.py

```diff
@@ -135,13 +135,42 @@ def request_session(
                     "reason": "The Model has no connected Git Model. Please contact a project manager or admininistrator",
                 },
             )
+
+        if (
+            body.branch is None
+            or not body.branch.startswith("refs/heads/")
+            and not body.branch.startswith("refs/tags/")
+            and not body.branch == "All"
+        ):
+            raise HTTPException(
+                status_code=400,
+                detail={
+                    "err_code": "branch_format_not_exists",
+                    "reason": f"Either branch {body.branch} not exists or has not the proper format.",
+                },
+            )
+
+        if body.depth == DepthType.LatestCommit:
+            depth = 1
+        elif body.depth == DepthType.CompleteHistory:
+            depth = 0
+        else:
+            raise HTTPException(
+                status_code=400,
+                detail={
+                    "err_code": "wrong_depth_format",
+                    "reason": f"Depth type {depth} is not allowed.",
+                },
+            )
```

Comment:
The type check is already done with the pydantic models - the code will never be executed. If the body does not match the model, fastAPI will return a bad request error.